perun-core/src/test/java/cz/metacentrum/perun/core/entry/ResourcesManagerEntryIntegrationTest.java

```diff
@@ -117,6 +117,49 @@ public void createResourceWhenFacilityNotExists() throws Exception {
 
 	}
 
+	@Test (expected=ResourceExistsException.class)
+	public void copyResourceWithExistingNameInDestinationFacility() throws Exception{
+		System.out.println(CLASS_NAME + "copyResourceWithExistingNameInDestinationFacility");
+
+		vo = setUpVo();
+		facility = setUpFacility();
+
+		Resource resource1 = new Resource();
+		resource1.setName("TestResource");
+		resource1.setDescription("Template Resource");
+
+		String newResourceName = "TestResource";
+
+		resource1 = resourcesManager.createResource(sess, resource1, vo, facility);
+		assertNotNull("unable to create resource1 before copying",resource1);
+
+		resourcesManager.copyResource(sess, resource1, newResourceName, vo, facility);
+	}
+
+	@Ignore //copyResource method is not completed yet.
```

Comment:
Test is ignored for now because i don't know if i can PR when test fails. The problem occurs when we call setAttribute in ResourcesManagerBlImpl (70th line). The method tries to copy ID of resource:att-def:core attribute and throws WrongAttributeAssignment exception. How do we solve this issue of copying IDs?

Reply:
You can't copy core attributes. Core attribute is for example 'id' of Object which is get by method "getId" on such Object, but it is returned as core attribute.